### ISSUE TITLE
[PRE REVIEW]: languagemodels: A Python Package for Exploring Modern Natural Language Processing

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#210]
**Submitting author:** @jncraton (<a href="http://orcid.org/0009-0007-6543-8571">Jonathan Craton</a>)
**Repository:** https://github.com/jncraton/languagemodels
**Branch with paper.md** (empty if default branch): 
**Version:** v0.0.4
**Editor:** @juanklopper
**Reviewers:** @maxale
**Managing EiC:** Lorena Barba

 

## Status

[![status](https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4/status.svg)](https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4"><img src="https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4/status.svg)](https://jose.theoj.org/papers/9cae85cfc3ac93b6a52a27b97ba77cd4)
```

**Author instructions**

Thanks for submitting your paper to JOSE @jncraton. **Currently, there isn't an JOSE editor assigned** to your paper.

@jncraton if you have any suggestions for potential reviewers then please mention them here in this thread (without tagging them with an @). In addition, [this list of people](https://bit.ly/jose-reviewers) have already agreed to review for JOSE and may be suitable for this submission (please start at the bottom of the list).

**Editor instructions**

The JOSE submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks.

**:warning: JOSE reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSE is currently operating in a "reduced service mode". 

For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Wordcount for `paper.md` is 756

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.03 s (487.9 files/s, 48273.7 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                           7            194            188            343
Markdown                         3             79              0            182
TeX                              1             20              0            174
Jupyter Notebook                 3              0            360             63
YAML                             2              4              0             39
make                             1              8              0             28
-------------------------------------------------------------------------------
SUM:                            17            305            548            829
-------------------------------------------------------------------------------


Statistical information for the repository '6da54c33ba74bc499aa4fcda' was
gathered on 2023/06/06.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Jon Craton                     158          1651            926          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Jon Craton                  725           43.9          0.4                5.52
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00210/jose.00210/10.21105.jose.00210.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00210/jose.00210/10.21105.jose.00210.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by xuanxu** [comment]
@editorialbot generate pdf

**Comment by labarba** [comment]
@editorialbot invite @juanklopper as editor

**Comment by labarba** [comment]
@jncraton 👋 
Let me start with an apology: I am the editor-in-chief and I have neglected JOSE for several months due to plain overwhelm. I have cut down on other service and am back here with intention of reviving our little alt-journal. Thank you for your submission! Given the time that has passed, do you have updates to the paper?

**Comment by jncraton** [comment]
@labarba No worries! I've made a few updates to the paper to cover some internal changes to the package. It should be ready for review now.

**Comment by jncraton** [comment]
@editorialbot generate pdf

**Comment by labarba** [comment]
@editorialbot invite @juanklopper as editor

**Comment by labarba** [comment]
@editorialbot invite @juanklopper as editor

**Comment by juanklopper** [comment]
@editorialbot assign me as editor

**Comment by juanklopper** [comment]
@editorialbot commands

**Comment by juanklopper** [comment]
@editorialbot add @maxale as reviewer

**Comment by labarba** [comment]
I have now sent a reminder to @juanklopper via email to complete the Pre-review by assigning a second reviewer. Stay tuned, and feel free to ping here if we drop it again. Volunteers are time poor!

**Comment by labarba** [comment]
Hi folks 👋  – I see that this submission has been stalled for a long time. I'm afraid it is not the only one.. we are struggling greatly to find volunteer reviewers for JOSE submissions, and currently we have 72 in-progress papers with only 13 having secured reviewers. Given that, I am commenting in open Pre-Reviews with a request for hopeful authors to help us out by volunteering to review each others' submissions! Please look at the [submissions in pre-review here](https://github.com/openjournals/jose-reviews/issues) and pick one to review! 🙏

**Comment by labarba** [comment]
@ayush9pandey 👋  – You signed up as a potential reviewer for The Journal of Open Source Education. Would you consider helping us with this submission? It's been stuck in Pre-review for a long time, and we're struggling because no reviewers in our database explicitly listed Large Language Models or NLP as their primary domain. 

**Paper Title**: languagemodels: Explore Large Language Models in 512MB of RAM
**Description**: This paper introduces a lightweight Python package designed to make large language models accessible to students and educators on low-resource hardware. By abstracting away complex deployment configurations and optimizing for low RAM footprints, it aims to lower the barrier to teaching modern NLP and generative AI concepts in introductory educational environments.

I found you via a "loose match" search with:
- Primary Programming Language: Python
- Domains of Expertise: Computer Science, Artificial Intelligence/Machine Learning, Computer Science Education

**Comment by labarba** [comment]
@jncraton 👋  – I've been sprinting on JOSE, trying to dig us out of a grizzly backlog. We are trying to get things moving. Feel free to make any updates to the paper, given all the time that has passed, if you have continued work on the software. Keep calm and carry on!

**Comment by labarba** [comment]
@editorialbot check repository

**Comment by ayush9pandey** [comment]
Sure! I can review this.

**Comment by labarba** [comment]
Awesome! Thank you @ayush9pandey 🙏

**Comment by labarba** [comment]
@maxale 👋  – You had agreed a while back to review this submission for The Journal of Open Souce Education. I realize a long time has passed: are you still available to help us out?
Here's a reminder of the context... 
**Paper Title**: languagemodels: Explore Large Language Models in 512MB of RAM
**Description**: This paper introduces a lightweight Python package designed to make large language models accessible to students and educators on low-resource hardware. By abstracting away complex deployment configurations and optimizing for low RAM footprints, it aims to lower the barrier to teaching modern NLP and generative AI concepts in introductory educational environments.

**Comment by labarba** [comment]
@editorialbot add @ayush9pandey as reviewer

**Comment by juanklopper** [comment]
@ayush9pandey thank you so much for taking the time to review this submission. @maxale I hope that you can help!!

**Comment by jncraton** [comment]
@ayush9pandey Thanks for taking the time to look this over. Since this review was started, I've used this package for a few learning modules that I've made available publicly: 

- <https://www.merlot.org/merlot/viewMaterial.htm?id=773418755>
- <https://clark.center/details/jlcraton/d44df918-bcfa-462a-9bf2-26e619719c69>

Hopefully that helps to get a sense for the target audience. I've tried to make this package very easy to use for folks that have limited technical experience and have used it successfully for several years in my CS1 courses.

Feel free to reach out if you have any questions.

**Comment by ayush9pandey** [comment]
I'll wait for the review issue to go up.

**Comment by labarba** [comment]
Yes, thank you! We just need to wait for confirmation from the first assigned reviewer, @maxale, since it was so long ago since they agreed to review.
